### PR TITLE
Add a format attribute to dt_print to allow gcc to check arguments

### PR DESCRIPTION
**Pull request by bruceg**
Also fix the resulting errors

### CODE REVIEW

File: src/develop/tiling.c [hunk 1]
```diff
@@ -1821,7 +1821,7 @@ static int _default_process_tiling_cl_roi(struct dt_iop_module_t *self, struct d
                      + ((size_t)oroi_good.x - roi_out->x) * out_bpp;
 
       dt_print(DT_DEBUG_OPENCL,
-               "[default_process_tiling_cl_roi] tile (%d, %d) with %d x %d at origin [%d, %d]\n", tx, ty,
+               "[default_process_tiling_cl_roi] tile (%ld, %ld) with %d x %d at origin [%d, %d]\n", tx, ty,
```
Comment (LebedevRI) [inline src/develop/tiling.c]: Makes sense, but, why `%ld`?
Clang suggests `%zu`, and IIRC that is what is used elsewhere to print size_t.
E.g.:
```
/home/lebedevri/darktable/src/develop/tiling.c:1824:103: error: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Werror,-Wformat]
               "[default_process_tiling_cl_roi] tile (%d, %d) with %d x %d at origin [%d, %d]\n", tx, ty,
                                                          ~~                                          ^~
                                                          %zu
```

Comment (bruceg) [inline src/develop/tiling.c]: I used %ld because that is what GCC suggested and I didn't know about %z until just now. I will fix that.

**Comment by LebedevRI** [comment]
I'll merge, though i'm pretty sure that it is not the last function that lacks format attribute.

**Comment by bruceg** [comment]
You would be right. A quick grep reveals other public variadic functions that are likely to use printf formats. I'll get them too.

**Pull request merged**